SUBMARINE-833. Support ranger 2.1

harsha-tenneti opened the pull request:
### What is this PR for?
Support Ranger 2.1 version

* Split ranger source dir ranger-2 to ranger-2.0 and ranger-2.1 to capture breaking api changes
* Move RangerSparkPlugin to individual ranger-{ranger.major.version} main directories and modify according to ranger version
* Move RangerAdminClientImpl to individual ranger-{ranger.major.version} test directories and modify according to ranger version
* Introduce new profile ranger-2.1 and include elastic search dependency

### What type of PR is it?
[Improvement | Feature ]

### Todos
* Split ranger source dir ranger-2 to ranger-2.0 and ranger-2.1 to capture breaking api changes
* Move RangerSparkPlugin to individual ranger-{ranger.major.version} main directories and modify according to ranger version
* Move RangerAdminClientImpl to individual ranger-{ranger.major.version} test directories and modify according to ranger version
* Introduce new profile ranger-2.1 and include elastic search dependency

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-833

### How should this be tested?
https://travis-ci.org/github/harsha-tenneti/submarine/builds/772933964
<!--
* First time? Setup Travis CI as described on https://submarine.apache.org/contribution/contributions.html#continuous-integration
* Strongly recommended: add automated unit tests for any new or changed behavior
* Outline any manual steps to test the PR here.
-->
### Screenshots (if appropriate)

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No

harsha-tenneti commented:
@yaooqinn can you please help look at this PR

pingsutw commented:
Gentle ping @yaooqinn. Could you take a look? Thanks.

yaooqinn commented on .travis.yml:
```diff
@@ -264,6 +264,16 @@ matrix:
         - PROFILE="-Pspark-2.3 -Pranger-2.0"
         - MODULES="-pl :submarine-spark-security"
 
+    - name: Test submarine spark security with spark 2.3 and ranger 2.1
```
we don't need to test spark 2.3

harsha-tenneti replied:
removed spark 2.3 test

yaooqinn commented on submarine-security/spark-security/pom.xml:
```diff
@@ -36,6 +36,7 @@
   <properties>
     <antlr4.version>4.7</antlr4.version>
     <eclipse.jpa.version>2.5.2</eclipse.jpa.version>
+    <es.version>7.10.2</es.version>
```
elasticsearch.version

harsha-tenneti replied:
modified

yaooqinn commented on submarine-security/spark-security/ranger-1/src/test/org.apache.submarine.spark.security/RangerAdminClientImpl.scala:
```diff
@@ -0,0 +1,62 @@
+  }
+
+  override def getServicePoliciesIfUpdated(
+                                            lastKnownVersion: Long,
```
indention

harsha-tenneti replied:
modified

yaooqinn commented on submarine-security/spark-security/ranger-1/src/test/org.apache.submarine.spark.security/RangerAdminClientImpl.scala:
```diff
@@ -0,0 +1,62 @@
+  override def revokeAccess(request: GrantRevokeRequest): Unit = {}
+
+  override def getServiceTagsIfUpdated(
+                                        lastKnownVersion: Long,
+                                        lastActivationTimeInMillis: Long): ServiceTags = null
```
ditto

harsha-tenneti replied:
modified

yaooqinn reviewed (approved):
LGTM, only minors

pingsutw commented:
@harsha-tenneti Any update?

harsha-tenneti commented:
Modified as per the suggestions

harsha-tenneti commented:
@yaooqinn made some changes as the tests are failing due to missed imports of RangerSParkPlugin in CreateRoleCommand, DropRoleCommand, ShowCurrentRolesCommand, and ShowRolesCommand classes. 

Verified locally that tests are passing. Please help review

kevin85421 commented:
@harsha-tenneti Thank you for your contribution! Recently, we deprecate Travis CI and use Github Action instead. Hence, there is a conflicting file ".travis.yml" in your patch. There are two solutions:

(1) Remove the ".travis.yml" file in your patch, and I will issue a new patch to help you run the test on Github Action

(2) Rebase your branch with the master branch, and modify the ["master.yml"](https://github.com/apache/submarine/blob/master/.github/workflows/master.yml#L482) to run the test on Github Action.

Both of the options are fine. Feel free to contact me if the patch is ready to be merge. Thanks!

harsha-tenneti commented:
@kevin85421 thanks for the info. I went with option 2 wherein I rebased with master and modified master.yml to include ranger 2.1. 

Please have a look and let me know if I am missing anything

kevin85421 commented:
Thank you @harsha-tenneti! It looks good to me. I will merge it.

kevin85421 commented:
@harsha-tenneti Can you email kaihsun@apache.org with your JIRA username? Submarine contributors need the privilege to create JIRA issues. With the privilege, you can create and assign JIRA issues. Thanks!
